source/common/router/config_impl.cc

```diff
@@ -952,6 +1057,11 @@ absl::optional<std::string> RouteEntryImplBase::currentUrlPathAfterRewriteWithMa
     }
   }
 
+  if (path_rewrite_policy_.enabled()) {
+    auto just_path(Http::PathUtil::removeQueryAndFragment(headers.getPathValue()));
+    return *std::move(path_rewrite_policy_.predicate()->rewriteUrl(just_path, matched_path));
```

Comment:
I believe calling `operator*` on the `absl::StatusOr` returned by `rewriteUrl` is undefined unless you've checked that `ok()` returns true. Even if it throws an exception, we don't allow that in the request path.

One option might be to say that the rewrite extension needs to handle errors and jsut return the unmodified URL. Otherwise we could handle it here. There's an argument that this should turn into an HTTP error, but I don't think we've done that for other cases.

Reply:
Handled within router. We return the unmodified path in case of failure